src/uwtools/drivers/driver.py

```diff
@@ -115,25 +117,32 @@ def _run_via_local_execution(self):
 
     @staticmethod
     def _create_user_updated_config(
-        config_class: Type[Config], config_values: dict, path: Path
+        config_class: Type[Config], config_values: dict, path: Path, schema: Optional[dict] = None
     ) -> None:
         """
         Create a config from a base file, user-provided values, or a combination of the two.
 
         :param config_class: The Config subclass matching the config type.
         :param config_values: The configuration object to update base values with.
         :param path: Path to dump file to.
+        :param schema: Schema to validate final config against.
         """
-        path.parent.mkdir(parents=True, exist_ok=True)
         user_values = config_values.get("update_values", {})
         if base_file := config_values.get("base_file"):
-            config_obj = config_class(base_file)
-            config_obj.update_values(user_values)
-            config_obj.dereference()
-            config_obj.dump(path)
+            cfgobj = config_class(base_file)
+            cfgobj.update_values(user_values)
+            cfgobj.dereference()
+            config = cfgobj.data
+            dump = partial(cfgobj.dump, path)
+        else:
+            config = user_values
+            dump = partial(config_class.dump_dict, config, path)
+        if validate(schema=schema or {"type": "object"}, config=config):
+            path.parent.mkdir(parents=True, exist_ok=True)
+            dump()
+            log.debug(f"Wrote config to {path}")
```

Comment:
This is the crux of the whole PR: If the config validates -- either using a passed-in schema or a default permissive schema -- _then_ make the destination directory and dump the config there. If the config fails to validate, don't write anything to disk and (below) log an error.